haas/client/errors.py

```diff
@@ -0,0 +1,22 @@
+class AuthenticationError(Exception):
+    pass
+
+
+class NotFoundError(Exception):
+    pass
+
+
+class DuplicateError(Exception):
+    pass
+
+
+class BlockedError(Exception):
+    pass
+
+
+class ProjectMismatchError(Exception):
```

Comment:
Now that you've refactored things to de-duplicate the error checking, I don't think you're using any of the errors in this file except for FailedAPICallException. In that case, you should get rid of the others, and probably move the remaining one to another module, just because it seems silly to have a whole file just for the one trivial error type.